virtualizarr/manifests/registry.py

```diff
@@ -0,0 +1,236 @@
+"""
+Based on https://docs.rs/object_store/0.12.2/src/object_store/registry.rs.html#176-218
+
+This module may eventually be upstreamed into https://github.com/developmentseed/obstore
+"""
+
+from __future__ import annotations
+
+from collections import namedtuple
+from typing import TYPE_CHECKING, Dict, Iterator, Optional, Tuple, TypeAlias
+from urllib.parse import urlparse
+
+if TYPE_CHECKING:
+    from obstore.store import (
+        ObjectStore,
+    )
+
+Url: TypeAlias = str
+Path: TypeAlias = str
+
+UrlKey = namedtuple("UrlKey", ["scheme", "netloc"])
+
+
+def get_url_key(url: Url) -> UrlKey:
+    """
+    Generate the UrlKey containing a url's scheme and authority/netloc that is used a the
+    primary key's in a [ObjectStoreRegsitry.map][virtualizarr.manifests.ObjectStoreRegistry.map]
+
+    Parameters
+    ----------
+    url
+        Url to generate a UrlKey from
+
+    Returns
+    -------
+        NamedTuple containing the Url's scheme and authority/netloc
+
+    Raises
+    ------
+    ValueError
+        If provided Url does not contain a scheme based on [urllib.parse.urlparse][]
+    """
+    parsed = urlparse(url)
+    if not parsed.scheme:
+        raise ValueError(
```

Comment:
So here we are requiring a scheme - do we also want to require a prefic too?

Reply:
I'm inclined to at a minimum leave prefix restrictions to a separate PR since it's easily separable from these changes and would open up per-scheme restrictions (e.g., I don't think it makes sense to require a netloc or path/prefix for memory stores.